components/RemediesList.js

```diff
@@ -2,12 +2,31 @@ import Image from "next/image";
 import Link from "next/link";
 import styled from "styled-components";
 import FavoriteButton from "./FavoriteButton";
+import SearchBar from "./SearchBar";
+import { useState } from "react";
 
 export default function RemediesList({ remedies, handleToggleFavorite }) {
+  const [searchQuery, setSearchQuery] = useState("");
+  const filteredRemedies = remedies.filter((remedy) =>
+    remedy.title.toLowerCase().includes(searchQuery.toLowerCase())
+  );
+
+  function handleSearch(query) {
+    setSearchQuery(query);
+  }
+
+  function handleClearSearchBar() {
+    setSearchQuery("");
+  }
```

Comment:
Consider using a single function to perform the search.

This function accepts any value as query, so it can handle both actions.
 1.  For search –  it sets the searchQuery to the input value.
```suggestion
function handleSearch(query) {
    setSearchQuery(query);
  }
```